gwctl/pkg/resourcediscovery/discoverer.go

```diff
@@ -444,6 +534,47 @@ func fetchHTTPRoutes(ctx context.Context, k8sClients *common.K8sClients, filter
 	return httpRouteList.Items, nil
 }
 
+// fetchHTTPRoutes fetches HTTPRoutes based on a filter.
+func (d Discoverer) fetchReferenceGrants(ctx context.Context, filter Filter) ([]gatewayv1beta1.ReferenceGrant, error) {
+	gvr := schema.GroupVersionResource{
+		Group:    defaultReferenceGrantGroupVersion.Group,
+		Version:  defaultReferenceGrantGroupVersion.Version,
+		Resource: "referencegrants",
+	}
+	// TODO(gauravkghildiyal): Uncomment once
+	// https://github.com/kubernetes-sigs/gateway-api/pull/3001 merges
+	// if d.PreferredReferenceGrantGroupVersion != (metav1.GroupVersion{}) {
+	// 	gvr.Version = d.PreferredReferenceGrantGroupVersion.Version
+	// }
+
+	if filter.Name != "" {
+		// Use Get call.
+		referenceGrantUnstructured, err := d.K8sClients.DC.Resource(gvr).Namespace(filter.Namespace).Get(ctx, filter.Name, metav1.GetOptions{})
```

Comment:
Why use unstructured/dynamic client when we don't need to? I get the rationale when we're working with arbitrary types, but this feels unnecessarily complicated. Is this primarily to enable using the preferred API version in a cluster?

Reply:
You guessed right! It's used to _slightly_ easily deal with fetching an arbitrary versioned resource.

With typed clients, it get slightly more messy.